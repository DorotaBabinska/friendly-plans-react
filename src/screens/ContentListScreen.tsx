import React from 'react';
import FastImage from 'react-native-fast-image';

import { FullScreenTemplate } from 'components';
import { i18n } from 'locale';
import { headerStyle } from 'styles';

export class ContentListScreen extends React.PureComponent {
  static navigationOptions = {
    title: i18n.t('content:contentList'),
    headerTitleStyle: headerStyle.headerText,
  };

  render() {
    const imageNo = Math.floor(Math.random() * 11);
    return (
      <FullScreenTemplate>
        <FastImage
          style={{ height: '100%' }}
          source={{
            uri: `https://unsplash.it/400/700?image=${imageNo}`,
            priority: FastImage.priority.normal,
          }}
          resizeMode={FastImage.resizeMode.cover}
        />
      </FullScreenTemplate>
    );
  }
}
