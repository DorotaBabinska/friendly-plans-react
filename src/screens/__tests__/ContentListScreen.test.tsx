import { shallow } from 'enzyme';
import React from 'react';

import { ContentListScreen } from '../ContentListScreen';

describe('ContentListScreen', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<ContentListScreen />);
    expect(wrapper).toMatchSnapshot();
  });
});
