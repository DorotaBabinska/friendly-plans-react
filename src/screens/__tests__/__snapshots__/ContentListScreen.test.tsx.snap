// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentListScreen should match snapshot 1`] = `
<FullScreenTemplate>
  <FastImage
    resizeMode="cover"
    source={
      Object {
        "priority": "normal",
        "uri": "https://unsplash.it/400/700?image=2",
      }
    }
    style={
      Object {
        "height": "100%",
      }
    }
  />
</FullScreenTemplate>
`;
